File: frontend/src/lib/api/content.ts
/**
 * Content API Client
 *
 * 콘텐츠, 카탈로그, 시리즈, 핸드 API
 */

import { api } from './client';

// =============================================================================
// Types
// =============================================================================

export interface Catalog {
	id: number;
	name: string;
	slug: string;
	description?: string;
	thumbnailUrl?: string;
	seriesCount?: number;
}

export interface Series {
	id: number;
	title: string;
	catalogId: number;
	year?: number;
	description?: string;
	thumbnailUrl?: string;
	contentCount?: number;
}

export interface Content {
	id: number;
	title: string;
	description?: string;
	seriesId?: number;
	durationSec: number;
	thumbnailUrl?: string;
	viewCount: number;
	createdAt: string;
	series?: {
		id: number;
		title: string;
		catalogId: string;
	};
}

export interface ContentDetail extends Content {
	files: ContentFile[];
	hands: Hand[];
}

export interface ContentFile {
	id: number;
	contentId: number;
	path: string;
	format: string;
	resolution?: string;
	sizeMb?: number;
}

export interface Hand {
	id: number;
	contentId: number;
	handNumber: number;
	timestampSec: number;
	players: HandPlayer[];
	potSize?: number;
	winner?: string;
	description?: string;
}

export interface HandPlayer {
	id: number;
	handId: number;
	playerId: number;
	playerName: string;
	position?: string;
	cards?: string;
	action?: string;
}

export interface ContentListResponse {
	items: Content[];
	total: number;
	page: number;
	hasNext: boolean;
}

// =============================================================================
// API Functions
// =============================================================================

/**
 * Get all catalogs
 */
export async function getCatalogs(): Promise<Catalog[]> {
	return api.get<Catalog[]>('/catalogs');
}

/**
 * Get catalog by ID
 */
export async function getCatalog(id: number): Promise<Catalog> {
	return api.get<Catalog>(`/catalogs/${id}`);
}

/**
 * Get series by catalog ID
 */
export async function getSeriesByCatalog(catalogId: number): Promise<Series[]> {
	return api.get<Series[]>(`/catalogs/${catalogId}/series`);
}

/**
 * Get series by ID
 */
export async function getSeries(id: number): Promise<Series> {
	return api.get<Series>(`/series/${id}`);
}

/**
 * Get contents with filters
 */
export async function getContents(options?: {
	seriesId?: number;
	catalogId?: number;
	page?: number;
	limit?: number;
	sort?: 'recent' | 'popular';
}): Promise<ContentListResponse> {
	const params = new URLSearchParams();
	if (options?.seriesId) params.set('series_id', options.seriesId.toString());
	if (options?.catalogId) params.set('catalog_id', options.catalogId.toString());
	if (options?.page) params.set('page', options.page.toString());
	if (options?.limit) params.set('limit', options.limit.toString());
	if (options?.sort) params.set('sort', options.sort);

	const query = params.toString();
	return api.get<ContentListResponse>(`/contents${query ? `?${query}` : ''}`);
}

/**
 * Get content by ID with details
 */
export async function getContent(id: number): Promise<ContentDetail> {
	return api.get<ContentDetail>(`/contents/${id}`);
}

/**
 * Get hands for a content
 */
export async function getHands(contentId: number): Promise<Hand[]> {
	return api.get<Hand[]>(`/contents/${contentId}/hands`);
}

// Export as namespace
export const contentApi = {
	getCatalogs,
	getCatalog,
	getSeriesByCatalog,
	getSeries,
	getContents,
	getContent,
	getHands
};

export default contentApi;

File: backend/src/schemas/content.py
"""
Content Schemas

콘텐츠 관련 Pydantic 스키마
"""

from datetime import datetime

from pydantic import BaseModel, Field

from .common import PaginatedResponse


class ContentResponse(BaseModel):
    """콘텐츠 응답"""

    id: int
    series_id: int = Field(alias="seriesId")
    episode_num: int | None = Field(None, alias="episodeNum")
    title: str
    description: str | None = None
    duration_sec: int = Field(alias="durationSec")
    thumbnail_url: str | None = Field(None, alias="thumbnailUrl")
    view_count: int = Field(alias="viewCount")
    hands_count: int = Field(0, alias="handsCount")
    created_at: datetime = Field(alias="createdAt")

    class Config:
        from_attributes = True
        populate_by_name = True


class SeriesInfo(BaseModel):
    """시리즈 정보 (콘텐츠 상세용)"""

    id: int
    catalog_id: str = Field(alias="catalogId")
    title: str
    year: int

    class Config:
        from_attributes = True
        populate_by_name = True


class HandSummary(BaseModel):
    """핸드 요약 (콘텐츠 상세용)"""

    id: int
    hand_number: int | None = Field(None, alias="handNumber")
    start_sec: int = Field(alias="startSec")
    end_sec: int = Field(alias="endSec")
    grade: str
    players: list[str]
    is_all_in: bool = Field(alias="isAllIn")

    class Config:
        from_attributes = True
        populate_by_name = True


class ContentDetailResponse(BaseModel):
    """콘텐츠 상세 응답"""

    id: int
    title: str
    description: str | None = None
    duration_sec: int = Field(alias="durationSec")
    thumbnail_url: str | None = Field(None, alias="thumbnailUrl")
    view_count: int = Field(alias="viewCount")
    episode_num: int | None = Field(None, alias="episodeNum")
    series: SeriesInfo
    stream_url: str | None = Field(None, alias="streamUrl")
    hands: list[HandSummary]
    hands_count: int = Field(alias="handsCount")

    class Config:
        from_attributes = True
        populate_by_name = True


class ContentListResponse(BaseModel):
    """콘텐츠 목록 응답"""

    items: list[ContentResponse]
    total: int
    page: int
    limit: int
    has_next: bool = Field(alias="hasNext")

    class Config:
        populate_by_name = True
